src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/RightCurlyCheck.java

```diff
@@ -235,8 +234,8 @@ private static boolean shouldBeAloneOnLineWithAloneOption(Details details,
     private static boolean shouldBeAloneOnLineWithAloneOrSinglelineOption(Details details,
                                                                           String targetSrcLine) {
         return !isAloneOnLine(details, targetSrcLine)
-                && !isSingleLineBlock(details)
-                && !isEmptyBody(details.lcurly);
+                && !isEmptyBody(details.lcurly)
+                && !isSingleLineBlock(details);
```

Comment:
Please explain this change.

Reply:
This is for the coverage. Previously, `isEmptyBody` is also used in `shouldBeAloneOnLineWithAloneOption`, so the coverage is guaranteed. 

Now, it is the only place for this method, so the order is important.

To cover one of line in  `isEmptyBody`, the following line is added.
`class TestClass4 { }`
However, `!isSingleLineBlock(details)` will return false in the first place,  so this line cannot be passed in the `isEmptyBody` function anymore.

Let's reorder them so that such line will be examined in the first place.

Reply:
This is no longer re-ordered so I am assume it was resolved so I will mark this as resolved.